sparkctl/cmd/create.go

```diff
@@ -267,28 +273,116 @@ func isContainerLocalFile(file string) (bool, error) {
 	return false, nil
 }
 
+type Blob interface {
+	setPublicACL(ctx context.Context, bucket string, filePath string) error
+}
+
+type BlobHandler struct {
+	ctx    context.Context
+	bucket *blob.Bucket
+	blob   Blob
+}
+
+func (bh BlobHandler) uploadToBucket(
+	uploadScheme string,
+	uploadBucket string,
+	uploadPath string,
+	localFilePath string) (string, error) {
+	fileName := filepath.Base(localFilePath)
+	uploadFilePath := filepath.Join(uploadPath, fileName)
+	// Check if exists by trying to fetch metadata
+	_, err := bh.bucket.NewRangeReader(bh.ctx, uploadFilePath, 0, 0)
+
+	if (blob.IsNotExist(err)) || (err == nil && Override) {
+		fmt.Printf("uploading local file: %s\n", fileName)
+
+		// Prepare the file for upload.
+		data, err := ioutil.ReadFile(localFilePath)
+		if err != nil {
+			return "", fmt.Errorf("Failed to read file: %s", err)
+		}
+
+		w, err := bh.bucket.NewWriter(bh.ctx, uploadFilePath, nil)
```

Comment:
Ditto. Need to make sure `w` gets closed, by adding `defer w.Close()` after the error check below.